pkg/cmd/pr/pr.go

```diff
@@ -101,108 +101,33 @@ func (o *Options) Run() error {
-			if o.Application == "" {
-				o.CommitTitle = fmt.Sprintf("chore(deps): upgrade to version %s", o.Version)
-			} else {
-				o.CommitTitle = fmt.Sprintf("chore(deps): upgrade %s to version %s", o.Application, o.Version)
-			}
-		}
+	// Auto-discover git URL and commit details if not provided
+	_, err = o.SetCommitDetails(o.Dir, o.CommitMessage, o.CommitTitle, o.Application)
+	if err != nil {
+		return fmt.Errorf("failed to set commit details: %w", err)
 	}
 
-	if o.AddChangelog != "" {
-		changelog, err := os.ReadFile(o.AddChangelog)
-		if err != nil {
-			return fmt.Errorf("failed to read changelog file %s: %w", o.AddChangelog, err)
-		}
-		o.EnvironmentPullRequestOptions.CommitChangelog = string(changelog)
+	// Handle changelog
+	err = o.SetChangeLog(o.AddChangelog)
+	if err != nil {
+		return fmt.Errorf("failed to set changelog: %w", err)
 	}
 
 	BaseBranchName := o.BaseBranchName
 
-	for i := range o.UpdateConfig.Spec.Rules {
-		rule := &o.UpdateConfig.Spec.Rules[i]
-		err = o.FindURLs(rule)
+	for i, rule := range o.UpdateConfig.Spec.Rules {
+		err = o.ProcessRule(&rule, i)
 		if err != nil {
-			return fmt.Errorf("failed to find URLs: %w", err)
+			return fmt.Errorf("failed to process rule #%d: %w", i, err)
 		}
 
-		o.Fork = rule.Fork
-		if len(rule.URLs) == 0 {
-			log.Logger().Warnf("no URLs to process for rule %d", i)
-		}
-		o.EnvironmentPullRequestOptions.SparseCheckoutPatterns = []string{}
-		if rule.SparseCheckout {
-			o.EnvironmentPullRequestOptions.SparseCheckoutPatterns, err = o.GetSparseCheckoutPatterns(*rule)
-			if err != nil {
-				return err
-			}
+		err = o.ProcessRuleURLs(&rule, BaseBranchName)
+		if err != nil {
+			return fmt.Errorf("failed to process URLs: %w", err)
 		}
-
-		for _, gitURL := range rule.URLs {
-			if gitURL == "" {
-				log.Logger().Warnf("missing out repository %d as it has no git URL", i)
-				continue
-			}
-
-			// lets clear the branch name so we create a new one each time in a loop
-			o.BranchName = ""
-			// lets reset the base branch name each time in a loop to avoid side effects when reusing pull requests
-			o.BaseBranchName = BaseBranchName
-
-			o.Function = func() error {
-				dir := o.OutDir
-
-				for _, ch := range rule.Changes {
-					err := o.ApplyChanges(dir, gitURL, ch)
-					if err != nil {
-						return fmt.Errorf("failed to apply change: %w", err)
-					}
-
-				}
-				return nil
-			}
-
-			if rule.ReusePullRequest {
-				if len(o.Labels) == 0 {
-					return fmt.Errorf("to be able to reuse pull request you need to supply pullRequestLabels in config file or --labels")
-				}
-				o.PullRequestFilter = &environments.PullRequestFilter{Labels: []string{}}
-				for _, label := range o.Labels {
-					o.PullRequestFilter.Labels = stringhelpers.EnsureStringArrayContains(o.PullRequestFilter.Labels, label)
-				}
-				if o.AutoMerge {
-					o.PullRequestFilter.Labels = stringhelpers.EnsureStringArrayContains(o.PullRequestFilter.Labels, environments.LabelUpdatebot)
-				}
-			}
```

Comment:
By removing this code you remove the functionality of reusing an existing open pull request. Please add it back.